designs/odcr.md

```diff
@@ -0,0 +1,213 @@
+
+Pros: 
+- Management of ODCRs usage is in hands of administrators (who are creating/updating EC2NodeClass configurations)
+  - Application developers are unaware and can use the existing `on-demand` and `spot` capacity types
+
+Cons:
+- ***TBD***
+
+#### Pricing and consolidation
+
+##### Provisioning
+
+Pricing is directly considered during provisioning and consolidation as capacity-reservation is prepaid. It is assumed to have a price of 0 during provisioning.
+
+##### Consolidating Capacity Reserved Instances
+
+During consolidation pricing does matter as it affects which candidate will be [prioritized](https://github.com/kubernetes-sigs/karpenter/blob/75826eb51589e546fffb594bfefa91f3850e6c82/pkg/controllers/disruption/consolidation.go#L156). Since all capacity instances are paid ahead of time, their cost is already incurred. Users would likely want to prioritize filling their reserved capacity.
+reservation first then fall back into other instances. Because of this reserved instances should likely show up as 0 dollar pricing when we calculate the instance pricing. Since each candidate is tied to a NodePool and EC2NodeClass, we should be able to safely override the pricing per node under a capacity reservation.
+
+##### Consolidating into Capacity Reserved Instances
+
+If we track Capacity Reservation usage, we can optimize the cluster configuration by moving non-Capacity Reserved instances into 
+Capacity Reserved instances. We would need to match the instance type, platform and availability zone prior to doing this.
+
+#### Labels
+
+When a node is launched against a CapacityReservation we will expose Capacity Reservation
+information as labels `karpenter.k8s.aws/capacity-reservation-id`. This is helpful for users to identify those nodes that are being used
+by a capacity reservation and the id of the reservation. Scenarios such as tracking how many nodes are under each capacity reservation and then
+checking how close to limit of the reservation.
+
+```yaml
+Name:               example-node
+Labels:             karpenter.k8s.aws/capacity-reservation-id=cr-12345
+                    karpenter.sh/capacity-type=on-demand
+```
+
+`karpenter.k8s.aws/capacity-reservation-id` will be the capacity reservation the node launched from. 
+
+We will propagate this information via [instance](https://github.com/aws/karpenter-provider-aws/blob/main/pkg/providers/instance/types.go#L29) by extracting it from [DescribeInstance](https://github.com/aws/karpenter-provider-aws/blob/main/pkg/batcher/describeinstances.go#L48) [aws doc]([https://docs.aws.amazon.com/sdk-for-go/api/service/ec2/#EC2.DescribeInstances](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_CapacityReservationSpecificationResponse.html)https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_CapacityReservationSpecificationResponse.html).
+
+### Failed to launch Nodes into Capacity Reservation
```

Comment:
I don't see us touching on a change in scheduling behavior to avoid more of these ICEs. Do we want to talk about a proposed change to capacity availability tracking where we track the number of available instance count in the offering vs. a binary yes/no about whether the offering has availability?